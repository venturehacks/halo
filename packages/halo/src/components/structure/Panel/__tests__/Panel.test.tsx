import { render } from 'enzyme';
import React from 'react';

import { Panel } from '../index';

describe('Panel', () => {
  test('smoke', () => {
    const component = render(<Panel>Content</Panel>);
    expect(component.text()).toContain('Content');
  });

  test('title', () => {
    const component = render(
      <Panel className="gap-10" title="Title">
        className
      </Panel>,
    );
    expect(component.hasClass('gap-10')).toBeTruthy();
  });

  test('className', () => {
    const component = render(<Panel className="gap-10">className</Panel>);
    expect(component.hasClass('gap-10')).toBeTruthy();
  });

  test('negativeSpace', () => {
    const component = render(
      <Panel negativeSpace="sm">Small negative space</Panel>,
    );
    expect(component.hasClass('p-6')).toBeTruthy();
    expect(component.hasClass('pt-4')).toBeTruthy();
  });

  test('flow', () => {
    const component = render(<Panel flow>Flow</Panel>);
    expect(component.hasClass('mb-6')).toBeTruthy();
  });

  describe('snapshot', () => {
    test('smoke', () => {
      const component = render(<Panel>Content</Panel>);

      expect(component).toMatchSnapshot();
    });

    test('title', () => {
      const component = render(<Panel title="General Settings">Content</Panel>);

      expect(component).toMatchSnapshot();
    });

    test('fully loaded props', () => {
      const component = render(
        <Panel className="gap-4" negativeSpace="sm" flow>
          Fully loaded props
        </Panel>,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
