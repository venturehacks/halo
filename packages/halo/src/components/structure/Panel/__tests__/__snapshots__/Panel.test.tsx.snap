// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel snapshot fully loaded props 1`] = `
<div
  class="flex flex-col border-t border-gray-100 rounded shadow-md bg-white max-w-3xl p-6 pt-4 mb-4 gap-4"
>
  Fully loaded props
</div>
`;

exports[`Panel snapshot smoke 1`] = `
<div
  class="flex flex-col border-t border-gray-100 rounded shadow-md bg-white max-w-3xl p-8 pt-6"
>
  Content
</div>
`;

exports[`Panel snapshot title 1`] = `
<div
  class="flex flex-col border-t border-gray-100 rounded shadow-md bg-white max-w-3xl p-8 pt-6"
>
  <header
    class="text-dark-aaaa font-medium antialiased text-lg mb-4"
  >
    General Settings
  </header>
  Content
</div>
`;
